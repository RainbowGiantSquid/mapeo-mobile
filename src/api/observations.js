// @flow
import { jsonRequest } from './base';
import { keyBy } from 'lodash';
import { parseObservationResponse } from '../models/observations';
import querystring from 'querystring'

import type {
  Observation as ObservationType,
  ServerObservationResponse,
  ServerObservationCreate
} from '../types/observation';

class Observation {
  static list = (opts) =>
    jsonRequest({
      method: 'GET',
      route: '/observations?' + querystring.stringify(opts)
    }).map(observations => {
      var payload = observations.map(parseObservationResponse)
      return keyBy(payload, 'id')
    });

  static create = (observation: ServerObservationCreate) =>
    jsonRequest({
      method: 'POST',
      route: '/observations',
      body: observation
    }).map(parseObservationResponse);

  static update = (observation: ServerObservationResponse) =>
    jsonRequest({
      method: 'PUT',
      route: `/observations/${observation.id}`,
      body: observation
    }).map(parseObservationResponse);
}

export default Observation;
