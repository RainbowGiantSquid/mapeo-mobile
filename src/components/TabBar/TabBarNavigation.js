import React from 'react';
import { TabNavigator } from 'react-navigation';
import MapViewNavigation from '@src/components/Views/MapView/MapViewNavigation';
import CameraViewNavigation from '@src/components/Views/CameraView/CameraViewNavigation';

const routeConfiguration = {
  MapViewNavigation: { screen: MapViewNavigation },
  CameraViewNavigation: { screen: CameraViewNavigation },
};

const tabConfiguration = {
  swipeEnabled: false,
  tabBarPosition: 'bottom',
  tabBarOptions: {
    activeTintColor: 'blue',
    showIcon: true,
    showLabel: false,
    style: {
      backgroundColor: 'white',
    },
  },
};

const TabBar = TabNavigator(routeConfiguration, tabConfiguration);

class TabBarNavigation extends React.Component {
 
  render() {
    return (
      <TabBar />
    );
  }
}

export default TabBarNavigation;
