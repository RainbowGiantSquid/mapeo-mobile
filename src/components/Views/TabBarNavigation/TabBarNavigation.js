// @flow
import React from 'react';
import { NavigationActions } from 'react-navigation';
import {
  StyleSheet,
  View,
  TouchableOpacity,
  ActivityIndicator,
  Text,
  Modal,
  Dimensions,
  ImageBackground,
  Image
} from 'react-native';
import Drawer from 'react-native-drawer';
import moment from 'moment';
import I18n from 'react-native-i18n';
import Icon from 'react-native-vector-icons/MaterialIcons';
import CollectionsImg from 'react-native-vector-icons/MaterialIcons';
import type { Observation } from '../../../types/observation';
import ObservationsView from '../ObservationsView';
import MapView from '../MapView';
import CameraView from '../CameraMainView';
import {
  WHITE,
  MAPEO_BLUE,
  LIGHT_GREY,
  VERY_LIGHT_BLUE,
  MEDIUM_GREY
} from '../../../lib/styles';
import CategoryPin from '../../../images/category-pin.png';
import Header from '../../Base/Header/Header';

const styles = StyleSheet.create({
  buttonText: {
    alignSelf: 'center',
    fontSize: 18,
    fontWeight: '400',
    color: MAPEO_BLUE
  },
  categoryContainer: {
    flex: 3,
    backgroundColor: VERY_LIGHT_BLUE,
    alignSelf: 'stretch',
    alignItems: 'center',
    borderRadius: 20
  },
  categoryPin: {
    width: 80,
    height: 90,
    marginTop: 15,
    justifyContent: 'center',
    alignItems: 'center'
  },
  confirmationModal: {
    alignSelf: 'center',
    alignItems: 'center',
    width: Dimensions.get('window').width * 0.8,
    height: Dimensions.get('window').height * 0.5,
    backgroundColor: 'white',
    borderRadius: 20
  },
  date: {
    color: MEDIUM_GREY,
    fontSize: 12,
    fontWeight: '400'
  },
  myObservationsIcon: {
    position: 'absolute',
    right: 20,
    top: 15
  },
  cameraIcon: {
    position: 'absolute',
    left: 20,
    top: 15
  },
  positionAtText: {
    fontSize: 12,
    color: 'black',
    fontWeight: '400'
  },
  positionText: {
    fontSize: 12,
    color: 'black',
    fontWeight: '700'
  },
  profileIcon: {
    position: 'absolute',
    left: 20,
    top: 15
  },
  savedContainer: {
    borderColor: LIGHT_GREY,
    borderBottomWidth: 1,
    flex: 1,
    alignSelf: 'stretch',
    justifyContent: 'center'
  },
  title: {
    fontSize: 20,
    fontWeight: '700',
    color: 'black',
    textAlign: 'center'
  }
});

type Props = {
  isFocused: boolean
};

export type StateProps = {
  selectedObservation: Observation,
  dispatch: any,
  navigation: NavigationActions
};

type State = {
  showModal: boolean,
  showCamera: boolean
};

I18n.fallbacks = true;
I18n.translations = {
  en: require('../../../translations/en'),
  es: require('../../../translations/es')
};

class TabBarNavigation extends React.Component<Props & StateProps, State> {
  state = {
    showModal: false,
    showCamera: false
  };

  componentDidMount() {
    this.setState({
      showModal: false
    });
  }

  componentWillReceiveProps(nextProps: Props & StateProps) {
    if (nextProps.isFocused) {
      const { navigation } = nextProps;
      const shouldShowModal = !!(
        navigation.state &&
        navigation.state.params &&
        navigation.state.params.showModal
      );
      if (shouldShowModal) {
        this.timeout = setTimeout(
          () =>
            this.setState({
              showModal: true
            }),
          2000
        );
      }
    }
  }

  shouldComponentUpdate(nextProps: Props & StateProps, nextState: State) {
    if (nextProps.isFocused) {
      if (nextProps !== this.props || nextState !== this.state) {
        return true;
      }
    }
    return false;
  }

  componentWillUnmount() {
    if (this.timeout) {
      clearTimeout(this.timeout);
    }
  }

  setModalVisible(visible: boolean) {
    this.setState({
      showModal: visible
    });
  }

  timeout: any;
  rightDrawer: Drawer;

  shouldShowModal() {
    const { navigation } = this.props;
    return !!(
      navigation.state &&
      navigation.state.params &&
      navigation.state.params.showModal
    );
  }

  closeRightDrawer = () => {
    this.rightDrawer.close();
  };

  openRightDrawer = () => {
    this.rightDrawer.open();
  };

  handleRightDrawerRef = (ref: Drawer) => {
    this.rightDrawer = ref;
  };

  goToCameraView = () => {
    this.setState({ showCamera: true });
  };

  goToMapView = () => {
    this.setState({ showCamera: false });
  };

  render() {
    const { dispatch, selectedObservation } = this.props;
    const { showModal, showCamera } = this.state;

    return (
      <Drawer
        ref={this.handleRightDrawerRef}
        content={<ObservationsView closeRightDrawer={this.closeRightDrawer} />}
        openDrawerOffset={0}
        side="right"
        type="displace"
      >
        <Header
          leftIcon={
            !showCamera ? (
              <TouchableOpacity onPress={this.goToCameraView}>
                <Icon color={WHITE} name="photo-camera" size={30} />
              </TouchableOpacity>
            ) : (
              <TouchableOpacity onPress={this.goToMapView}>
                <Icon color={WHITE} name="map" size={30} />
              </TouchableOpacity>
            )
          }
          rightIcon={
            <TouchableOpacity onPress={this.openRightDrawer}>
              <CollectionsImg color={WHITE} name="collections" size={30} />
            </TouchableOpacity>
          }
          style={{
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            zIndex: 5
          }}
        />
        {selectedObservation && (
          <Modal
            animation="slide"
            transparent
            visible={this.state.showModal}
            onRequestClose={() => {}}
          >
            <View
              style={{
                backgroundColor: 'rgba(52, 52, 52, 0.8)',
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center'
              }}
            >
              <View style={styles.confirmationModal}>
                <View style={styles.savedContainer}>
                  <Text
                    style={{
                      textAlign: 'center',
                      fontSize: 20,
                      fontWeight: '700',
                      color: 'black'
                    }}
                  >
                    {I18n.t('saved')}
                  </Text>
                </View>
                <View style={styles.categoryContainer}>
                  <ImageBackground
                    source={CategoryPin}
                    style={styles.categoryPin}
                  >
                    {selectedObservation && (
                      <View style={{ marginTop: -10 }}>
                        {selectedObservation.icon && (
                          <Image
                            source={selectedObservation.icon}
                            style={{ width: 30, height: 30 }}
                            resizeMode="contain"
                          />
                        )}
                      </View>
                    )}
                  </ImageBackground>
                  <View
                    style={{
                      justifyContent: 'center',
                      alignItems: 'center',
                      paddingBottom: 10
                    }}
                  >
                    <Text style={styles.title}>{selectedObservation.type}</Text>
                    <View style={{ flexDirection: 'row' }}>
                      <Text style={styles.positionAtText}>{I18n.t('at')} </Text>
                      <Text style={styles.positionText}>
                        {`${selectedObservation.lat}, ${
                          selectedObservation.lon
                        }.`}
                      </Text>
                    </View>
                    <Text style={styles.date}>
                      {I18n.t('on')}{' '}
                      {moment(selectedObservation.created).format(
                        'MMMM D, h:mm A'
                      )}
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        )}
        {!showCamera && <MapView />}
        {showCamera && <CameraView />}
      </Drawer>
    );
  }
}

export default TabBarNavigation;
