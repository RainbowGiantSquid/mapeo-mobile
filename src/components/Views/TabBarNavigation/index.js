// @flow
import { connect } from 'react-redux';
import { withNavigationFocus } from 'react-navigation';

const TabBarNavigation = require('./TabBarNavigation');
// import TabBarNavigation from './TabBarNavigation';

const mapStateToProps = state => ({
  navigationState: state.tabBar,
  selectedObservation: state.app.selectedObservation
});

export default connect(mapStateToProps)(withNavigationFocus(TabBarNavigation));
