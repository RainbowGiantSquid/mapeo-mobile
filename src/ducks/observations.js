// @flow
import { create } from '@src/lib/redux';
import update from 'immutability-helper';
import { keyBy } from 'lodash';
import { createSelector } from '@src/lib/selector';
import { StoreState } from '@types/redux';
import { Observation } from '@types/observation';

export const {
  type: OBSERVATION_LIST,
  action: observationList,
  reducer: observationListReducer,
} = create('OBSERVATION_LIST', {
  success: (state, action) => {
    const newState = update(state, {
      observations: { $set: keyBy(action.payload, 'id') },
    });

    return newState;
  },
});

export const selectContact = createSelector(
  [
    (state: StoreState, id: string): Observation => state.app.observations[id],
  ],
  (observation: Observation): Observation => observation,
  (observation: Observation, id: string): string => id,
);

export default [observationListReducer];
