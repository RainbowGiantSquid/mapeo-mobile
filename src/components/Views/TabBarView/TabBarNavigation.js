// @flow
import React from 'react';
import { addNavigationHelpers } from 'react-navigation';
import { StyleSheet, View, TouchableHighlight } from 'react-native';
import Drawer from 'react-native-drawer';
import { createReduxBoundAddListener } from 'react-navigation-redux-helpers';
import CollectionsImg from 'react-native-vector-icons/MaterialIcons';
import MyObservationsView from '../../Views/MyObservationsView';
import TabBar from './TabBar';
import { WHITE } from '../../../lib/styles';

const styles = StyleSheet.create({
  myObservationsIcon: {
    position: 'absolute',
    right: 20,
    top: 15
  },

  profileIcon: {
    position: 'absolute',
    left: 20,
    top: 15
  }
});

export type StateProps = {
  navigationState: any,
  dispatch: any
};

class TabBarNavigation extends React.Component<StateProps> {
  static router = TabBar.router;
  rightDrawer: Drawer;

  closeRightDrawer = () => {
    this.rightDrawer.close();
  };

  openRightDrawer = () => {
    this.rightDrawer.open();
  };

  handleRightDrawerRef = (ref: Drawer) => {
    this.rightDrawer = ref;
  };

  render() {
    const { dispatch, navigationState } = this.props;

    return (
      <Drawer
        ref={this.handleRightDrawerRef}
        content={
          <MyObservationsView closeRightDrawer={this.closeRightDrawer} />
        }
        openDrawerOffset={0}
        side="right"
        type="displace"
      >
        <View
          style={{
            flexDirection: 'row',
            height: 60,
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            zIndex: 5
          }}
        >
          <TouchableHighlight
            onPress={this.openRightDrawer}
            style={styles.myObservationsIcon}
            underlayColor="transparent"
          >
            <CollectionsImg color={WHITE} name="collections" size={40} />
          </TouchableHighlight>
        </View>
        <TabBar
          navigation={addNavigationHelpers({
            dispatch,
            state: navigationState,
            addListener: createReduxBoundAddListener('tabBar')
          })}
        />
      </Drawer>
    );
  }
}

export default TabBarNavigation;
