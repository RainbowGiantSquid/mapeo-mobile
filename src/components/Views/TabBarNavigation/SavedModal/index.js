// @flow
import { connect } from 'react-redux';
import { withNavigationFocus } from 'react-navigation';

import SavedModal from './SavedModal';
import { modalHide } from '../../../../ducks/modals';

const mapStateToProps = state => ({
  selectedObservation: state.app.selectedObservation
});

const mapDispatchToProps = dispatch => ({
  onHide: () => dispatch(modalHide('saved'))
});

export default connect(mapStateToProps, mapDispatchToProps)(SavedModal);
