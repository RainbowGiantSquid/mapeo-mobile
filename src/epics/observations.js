// @flow
import React from 'react';
import { Image } from 'react-native';
import type { ActionsObservable } from 'redux-observable';
import { OBSERVATION_LIST, observationList } from '../ducks/observations';
import { Action } from '../types/redux';
import { Observation } from '../types/observation';

export const observationListEpic = (
  action$: ActionsObservable<Action<string, Observation[]>>,
  store: any
) =>
  action$
    .ofType(OBSERVATION_LIST)
    .filter(
      action =>
        action.status === 'Start' &&
        store.getState().app &&
        store.getState().app.observations &&
        !store.getState().app.observations.length
    )
    .map(() =>
      observationList('', [
        {
          type: 'Old houses and farms',
          id: 1,
          lat: 0.106831547,
          lon: -77.42499473,
          link: 'link',
          created: new Date('2018-03-25T12:00:00Z'),
          name: 'Old houses and farms',
          notes: `Pedro and Maria's house`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/1-house.jpg')
            }
          ],
          icon: require('../images/categories/category_0.png'),
          mock: true
        },
        {
          type: 'Hunting sites',
          id: 2,
          lat: 0.088170932,
          lon: -77.44020885,
          link: 'link',
          created: new Date('2018-03-25T12:30:00Z'),
          name: 'Hunting sites',
          notes: `Don Pancho's hunting camp`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/4-camp.jpg')
            }
          ],
          icon: require('../images/categories/category_2.png'),
          mock: true
        },
        {
          type: 'Threat',
          id: 3,
          lat: 0.112411067,
          lon: -77.41378756,
          link: 'link',
          created: new Date('2018-02-03T17:30:00Z'),
          name: 'Threat',
          notes: `Oil spill 3 February 2018. Ruptured pipeline. Dead fish, oily water`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/5-threat.jpg')
            }
          ],
          icon: null,
          mock: true
        },
        {
          type: 'Medicinal plants',
          id: 4,
          lat: 0.09749025,
          lon: -77.43614404,
          link: 'link',
          created: new Date('2018-02-04T17:30:00Z'),
          name: 'Medicinal plants',
          notes: `Salola fruit plant - used in tea to heal inflamation`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/6-plant.jpg')
            }
          ],
          icon: require('../images/categories/category_6.png'),
          mock: true
        },
        {
          type: 'Waterfalls',
          id: 5,
          lat: 0.105818148,
          lon: -77.4408037,
          link: 'link',
          created: new Date('2018-03-04T17:30:00Z'),
          name: 'Waterfalls',
          notes: `Sacred waterfall, visited during vision quests for ancestors, for taking ayahuasca`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/7-waterfall.jpg')
            }
          ],
          icon: require('../images/categories/category_13.png'),
          mock: true
        },
        {
          type: 'Rivers and streams',
          id: 6,
          lat: 0.119004333,
          lon: -77.42374225,
          link: 'link',
          created: new Date('2018-03-24T17:30:00Z'),
          name: 'Rivers and streams',
          notes: `Point where pipeline crosses river`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/8-threat.jpg')
            }
          ],
          icon: require('../images/categories/category_0.png'),
          mock: true
        },
        {
          type: 'Historical sites',
          id: 7,
          lat: 0.082856811,
          lon: -77.4429836,
          link: 'link',
          created: new Date('2018-03-24T12:30:00Z'),
          name: 'Historical sites',
          notes: `There used to be a village here in the 1940s, it was abandoned after an epidemic of flu which killed most of the elders and children.`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/9-specialsite.jpg')
            }
          ],
          icon: require('../images/categories/category_17.png'),
          mock: true
        },
        {
          type: 'Gold',
          id: 8,
          lat: 0.108605553,
          lon: -77.41560494,
          link: 'link',
          created: new Date('2018-03-11T17:30:00Z'),
          name: 'Gold',
          notes: `Illegal gold miners 11 March 2018, excavators, dredges, open tailing pond, mercury.`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/10-threat.jpg')
            }
          ],
          icon: require('../images/categories/category_8.png'),
          mock: true
        }
      ])
    );

export default [observationListEpic];
