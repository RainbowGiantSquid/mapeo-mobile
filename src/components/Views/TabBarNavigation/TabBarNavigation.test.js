// @flow
import React from 'react';
import 'react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import TabBarNavigation from './TabBarNavigation';
import { createObservation } from '../../../mocks/observations';
import { createInitialStore } from '../../../lib/store';

jest.mock('../ObservationsView', () => () => null);
jest.mock('../MapView', () => () => null);
jest.mock('../CameraMainView', () => () => null);

describe('TabBarNavigation tests', () => {
  test('snapshots', () => {
    const observation = createObservation();
    const dispatch = jest.fn();
    const mockStore = configureStore();
    const state = {
      app: createInitialStore(),
      mainStack: []
    };

    const tree = renderer
      .create(
        <TabBarNavigation
          isFocused
          selectedObservation={observation}
          dispatch={dispatch}
          navigation={{ navigate: jest.fn() }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
