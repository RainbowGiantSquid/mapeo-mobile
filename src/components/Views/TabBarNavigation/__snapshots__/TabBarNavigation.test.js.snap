// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarNavigation tests snapshots 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    elevation={0}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "borderWidth": 0,
          "position": "absolute",
          "right": -750,
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "borderWidth": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "paddingHorizontal": 15,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 5,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, .8)",
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 35,
              "paddingLeft": 13,
              "paddingRight": 15,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#7AFA4C",
                "borderRadius": 50,
                "height": 10,
                "width": 10,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "marginHorizontal": 20,
              }
            }
          >
            +/- 100m
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <Modal
      animation="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(52, 52, 52, 0.8)",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 20,
              "height": 667,
              "width": 600,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "borderBottomWidth": 1,
                "borderColor": "#EAEAEA",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "textAlign": "center",
                }
              }
            >
              Saved!
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#ccffff",
                "borderRadius": 20,
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 90,
                  "justifyContent": "center",
                  "marginTop": 15,
                  "width": 80,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/category-pin.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 90,
                      "width": 80,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "marginTop": -10,
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/images/categories/category_18.png",
                    }
                  }
                  style={
                    Object {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingBottom": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 20,
                    "fontWeight": "700",
                    "textAlign": "center",
                  }
                }
              >
                Observation
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 12,
                      "fontWeight": "400",
                    }
                  }
                >
                  at
                   
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 12,
                      "fontWeight": "700",
                    }
                  }
                >
                  50.5, 50.5.
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666666",
                    "fontSize": 12,
                    "fontWeight": "400",
                  }
                }
              >
                on
                 
                March 25, 7:30 PM
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
